### PR TITLE
Add more es testcases

### PR DESCRIPTION
**Pull request by mkleen**
## Summary of the changes / Why this improves CrateDB

This ports back some elasticsearch testcases, which worked out of the box. It is taken from the state of elasticsearch 704317da71c. See commits for details.

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by mkleen** [comment]
retest this please

**Comment by mkleen** [comment]
retest this please

File: server/src/test/java/org/elasticsearch/cluster/ESAllocationTestCase.java [hunk 1]
```diff
@@ -144,11 +217,6 @@ public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, Routing
         public Decision canAllocate(ShardRouting shardRouting, RoutingAllocation allocation) {
             return decision;
         }
-
-        @Override
-        public Decision canAllocate(RoutingNode node, RoutingAllocation allocation) {
```
Comment (seut) [inline server/src/test/java/org/elasticsearch/cluster/ESAllocationTestCase.java:149]: why is this removed? it is not on the referred ES state https://github.com/elastic/elasticsearch/blame/704317da71c3de1b85679bc549bba3b4230acb5d/test/framework/src/main/java/org/elasticsearch/cluster/ESAllocationTestCase.java#L228
Seems like this could result in wrong behaviour.

Comment (mkleen) [reply server/src/test/java/org/elasticsearch/cluster/ESAllocationTestCase.java]: Good pick, i took it from the wrong point-of-time (head). My mistake. Thanks.

Comment (mkleen) [reply server/src/test/java/org/elasticsearch/cluster/ESAllocationTestCase.java]: I merged it already in the commit.

**Review by seut — APPROVED** [review]
👍
